scheduler/pkg/klcpermit/workflow_manager_test.go

```diff
@@ -73,3 +88,302 @@ func Test_getCRDName(t *testing.T) {
+			want1:               true,
+		},
+		{
+			name: "Test if secondary annotation is returned from annotations",
+			pod: &corev1.Pod{
+				ObjectMeta: metav1.ObjectMeta{
+					Annotations: map[string]string{
+						K8sRecommendedAppAnnotations: "some-app-name",
+					},
+				},
+			},
+			primaryAnnotation:   AppAnnotation,
+			secondaryAnnotation: K8sRecommendedAppAnnotations,
+			want:                "some-app-name",
+			want1:               true,
+		},
+		{
+			name: "Test if primary annotation is returned from labels",
+			pod: &corev1.Pod{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: map[string]string{
+						AppAnnotation: "some-app-name",
+					},
+				},
+			},
+			primaryAnnotation:   AppAnnotation,
+			secondaryAnnotation: K8sRecommendedAppAnnotations,
+			want:                "some-app-name",
+			want1:               true,
+		},
+		{
+			name: "Test if secondary annotation is returned from labels",
+			pod: &corev1.Pod{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: map[string]string{
+						K8sRecommendedAppAnnotations: "some-app-name",
+					},
+				},
+			},
+			primaryAnnotation:   AppAnnotation,
+			secondaryAnnotation: K8sRecommendedAppAnnotations,
+			want:                "some-app-name",
+			want1:               true,
+		},
+		{
+			name: "Test that empty string is returned when no annotations or labels are found",
+			pod: &corev1.Pod{
+				ObjectMeta: metav1.ObjectMeta{
+					Annotations: map[string]string{
+						"some-other-annotation": "some-app-name",
+					},
+				},
+			},
+			primaryAnnotation:   AppAnnotation,
+			secondaryAnnotation: K8sRecommendedAppAnnotations,
+			want:                "",
+			want1:               false,
+		},
+		{
+			name: "Test that empty string is returned when primary annotation cannot be found and secondary annotation is empty",
+			pod: &corev1.Pod{
+				ObjectMeta: metav1.ObjectMeta{
+					Annotations: map[string]string{
+						"some-other-annotation": "some-app-name",
+					},
+				},
+			},
+			primaryAnnotation:   AppAnnotation,
+			secondaryAnnotation: "",
+			want:                "",
+			want1:               false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got, got1 := getLabelOrAnnotation(tt.pod, tt.primaryAnnotation, tt.secondaryAnnotation)
+			if got != tt.want {
+				t.Errorf("getLabelOrAnnotation() got = %v, want %v", got, tt.want)
+			}
+			if got1 != tt.want1 {
+				t.Errorf("getLabelOrAnnotation() got1 = %v, want %v", got1, tt.want1)
+			}
+		})
+	}
+}
+
+func Test_getSpan_unbindSpan(t *testing.T) {
+	pod := &corev1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Labels: map[string]string{
+				WorkloadAnnotation: "myworkload",
+				VersionAnnotation:  "0.0.1",
+				AppAnnotation:      "myapp",
+			},
+		},
+	}
+
+	r := &WorkloadManager{
+		bindCRDSpan: make(map[string]trace.Span, 100),
+		Tracer:      trace.NewNoopTracerProvider().Tracer("trace"),
+	}
+
+	_, span := r.getSpan(context.TODO(), &unstructured.Unstructured{}, pod)
```

Comment:
Please, add comments on the different "phases" of the test. This way it is easier to immediately get what we are testing
For example, here "creating a single span if we invoke multiple time". Afterwards, "create new span with a new version" etc.